### ISSUE TITLE
Relation-Weighted Loss for Imbalanced Edge Types

### ISSUE THREAD

**Issue by shackett** [napistu/napistu-torch#91]
### Summary
Implement relation-weighted loss to handle imbalanced edge types (e.g., 80% "interaction" vs 2% "catalysis").

### Implementation
Add `relation_weight_alpha` config parameter that interpolates between uniform (0.0) and inverse frequency (1.0) weighting:
```python
def get_relation_weights(relation_counts: torch.Tensor, alpha: float = 0.5) -> torch.Tensor:
    """Compute normalized relation weights."""
    weights = 1.0 / (relation_counts ** alpha)
    return weights / weights.sum()

# In loss computation:
edge_weights = relation_weights[batch.relation_type]  # [num_edges]
loss = F.binary_cross_entropy_with_logits(scores, labels, weight=edge_weights)
```

### Config
```yaml
loss:
  relation_weight_alpha: 0.5  # 0.0=uniform, 0.5=sqrt, 1.0=inverse_freq
```

### Testing
- [ ] Weights sum to 1.0
- [ ] Alpha=0.0 equivalent to no weighting
- [ ] Log per-relation metrics to W&B
